src/main/java/net/openhft/chronicle/wire/WireMarshaller.java

```diff
@@ -629,8 +631,8 @@ protected void readValue(Object o, Object defaults, ValueIn read, boolean overwr
                 long pos = read.wireIn().bytes().readPosition();
                 try {
                     setValue(o, read, overwrite);
-                } catch (UnexpectedFieldHandlingException e) {
-                    Jvm.rethrow(e);
+                } catch (FatalMarshallingException e) {
+                    throw e;
```

Comment:
This is a bit nasty but we need a way for some exceptions to propagate through this catch-all stuff.

Reply:
I assume FatalMarshallingException extends IllegalStateException?

Reply:
I ended up canning this in favour of a more minimal solution